### ISSUE TITLE
mobx on strict mode

### ISSUE THREAD

**Issue by titoBouzout** [ryansolid/mobx-jsx#22]
I configure mobx to use strict mode as the documentation says enables better debugging and also provides great performance benefices. https://mobx.js.org/refguide/action.html

> actions will batch mutations and only notify computed values and reactions after the (outer most) action has finished

```javascript

mobx.configure({
	// https://mobx.js.org/refguide/api.html
	enforceActions: 'always',
	reactionScheduler: requestAnimationFrame,
})
```

So this means that any function that changes an observable should be wrapped in an action as `action(function(){ observable = new_value})`

Testing the other issue about the memo function, I noticed that mobx will throw errors as it seems that the memo function is changing an observable and is not wrapped in an action.

I always wrap my changes to observables in actions, but this library is doing changes to observables in unwrapped places, Im not completely sure why enforceActions is disabled by default in mobx as they strongly suggest to have this enable. Im also not sure whats the best approach to fix this on the library. I just tested disabling enforceAction while also having my changes wrapped in an action and seems to work just fine, so maybe would be possible to wrap any changes to observables in actions? Thanks again!

**Comment by ryansolid** [comment]
Yeah I can. It might just be the misunderstanding of actions, but I would have thought an autorun or computed that wrote to an observable would automatically be part of an action. But I say this from having written a number of libraries like MobX and not having looked into MobX specifically. And now on confirmation I can see they don't. This convinces me I should be looking at using computeds for memos and like things where possible. There area couple properties of computed that are undesirable here but I believe they should be worked around. I will work through this a bit.

**Comment by ryansolid** [comment]
Ok fixed for strict mode in 0.11.7. Could use computed due to need to handle nested disposal. Basically from MobX view I have a bunch of side effects. Re-open if you find any more cases I missed but I believe I wrapped all setters in actions.

**Comment by titoBouzout** [comment]
Thanks!
